### DISCUSSION TITLE
How to use one y-label in a facet grid

### DISCUSSION THREAD

**Discussion by mcp292** [vega/altair#2666] [Q&A] [answered]
Coming here as advised in https://github.com/altair-viz/altair-viz.github.io/issues/28. I will repeat everything here for convenience:

Using one label for an entire facet grid is not documented in [Adjusting Axis Labels](https://altair-viz.github.io/user_guide/customization.html#adjusting-axis-labels).

How can I make this plot only use one y-label?

# MWE
```py
import pandas as pd
import altair as alt


data = [{"P": "HD", "T": 0,  "C":100, "Max": 1},
        {"P": "HD", "T": 0,  "C":400, "Max": 3},
        {"P": "HD", "T": 0,  "C":800, "Max": 0},
        {"P": "HD", "T": 23, "C":100, "Max": 2},
        {"P": "HD", "T": 23, "C":400, "Max": 1},
        {"P": "HD", "T": 23, "C":800, "Max": 1},
        {"P": "HD", "T": 80, "C":100, "Max": 7},
        {"P": "HD", "T": 80, "C":400, "Max": 2},
        {"P": "HD", "T": 80, "C":800, "Max": 1},
        {"P": "LD", "T": 0,  "C":100, "Max": 1},
        {"P": "LD", "T": 0,  "C":400, "Max": 2},
        {"P": "LD", "T": 0,  "C":800, "Max": 7},
        {"P": "LD", "T": 23, "C":100, "Max": 7},
        {"P": "LD", "T": 23, "C":400, "Max": 1},
        {"P": "LD", "T": 23, "C":800, "Max": 0},
        {"P": "LD", "T": 80, "C":100, "Max": 2},
        {"P": "LD", "T": 80, "C":400, "Max": 0},
        {"P": "LD", "T": 80, "C":800, "Max": 1}]

df = pd.DataFrame(data)

chart = alt.Chart(df).mark_bar().encode(
    x="P:N",
    y=alt.Y("Max:Q", title="Maximum Long Word Observed (units)"),
    color="P:N",
    row="T:N",
    column="C:N"
).properties(width=100, height=100)
```

```py
     P   T    C  Max
0   HD   0  100    1
1   HD   0  400    3
2   HD   0  800    0
3   HD  23  100    2
4   HD  23  400    1
5   HD  23  800    1
6   HD  80  100    7
7   HD  80  400    2
8   HD  80  800    1
9   LD   0  100    1
10  LD   0  400    2
11  LD   0  800    7
12  LD  23  100    7
13  LD  23  400    1
14  LD  23  800    0
15  LD  80  100    2
16  LD  80  400    0
17  LD  80  800    1
```

![facet_grid](https://user-images.githubusercontent.com/49123398/182773758-c26c371d-2e37-4725-ba40-e4e76bdeb011.png)

**Comment by joelostblom** [comment]
You can see a couple of approaches for this here https://stackoverflow.com/questions/71600310/make-collective-title-for-columned-altair-chart

**Reply by mcp292** [reply]
This is actually very workable thank you! I was hoping to change this on the label level though, not add a title. Is this possible to configure?

**Reply by mcp292** [reply]
I would still want the T as the outermost label on the y, labeling the 0, 23, 80 in my graph above.

**Reply by joelostblom** [reply]
You could place a free standing mark text as in one of the answers on the link above. Other than that, it might be possible to add newlines to you `T` title so that the second line ends up under the `0`, `23`, `80`. Something like `['T', '', '', '', 'Maximum Long Work Hours']`. Not sure if that will work

**Reply by mcp292** [reply]
Wow! It is not documented that you can pass an iterable to either [header](https://altair-viz.github.io/user_guide/generated/core/altair.Header.html) or [title](https://altair-viz.github.io/user_guide/generated/channels/altair.Row.html)! This seems to work and is a very clever idea! See my answer below.

**Comment by mcp292** [comment, marked as answer]
Thanks to @joelostblom and [his comment](https://github.com/altair-viz/altair/discussions/2666#discussioncomment-3338269), this is one working solution:
```py
chart = alt.Chart(df).mark_bar().encode(
    x="P:N",
    y=alt.Y("Max:Q", title=None),
    color="P:N",
    row=alt.Row("T:N", title=["Maximum Long Word Observed (units)", "", "T"]),
    column="C:N"
).properties(width=100, height=100)
```

![one y-label](https://user-images.githubusercontent.com/49123398/183539327-407a7d9a-aa96-433b-aeb2-c8e2408eab34.png)

This seems to work, [but is not documented](https://github.com/altair-viz/altair/discussions/2666#discussioncomment-3353848). If there becomes a built-in I would love to know. Or a more low-level answer that actually makes the y-axis one, that would be great and more consistent with the other plots in the paper. Therefore, I will leave this answer open unless otherwise advised.

Also, the fact that defining `title` in `altair.Row()` overrides the facet header (`T`) is not documented. Unless "axis/header/legend" is removed, it says this is true in the opposite case:

> 2) If both field definition’s title and axis, header, or legend title are defined, axis/header/legend title will be used.

[Source](https://altair-viz.github.io/user_guide/generated/channels/altair.Row.html)

**Reply by mcp292** [reply]
The other problem with this is that the label is not in the correct position, up against the y-axis before the facet header labels, leaving room for misinterpretation.

**Reply by mcp292** [reply]
The ideal would be to edit this using [top-level chart configuration](https://altair-viz.github.io/user_guide/configuration.html) such as `chart.configure_header()`; after the fact, not within the encoding definition.

**Reply by joelostblom** [reply]
Regarding having the label closer to the y-axis. You can probably play around with the padding parameters of both the facet header and the facet labels (80, 23, 0) to fix this, but it might take some fiddling.

**Reply by mcp292** [reply]
Sorry, not that it should be closer as-is, but that it should be interwoven as in the original post: y-axis scale, y-axis label, [facet header label, facet header title](https://vega.github.io/vega-lite/docs/header.html).
